### PR TITLE
Small set of changes towards perf improvements

### PR DESCRIPTION
**Pull request by mjp41**

### CODE REVIEW

File: src/internal.cc [hunk 1]
```diff
@@ -256,13 +256,28 @@ namespace rego
     return in_query(node->parent()->shared_from_this());
   }
 
-  Node err(NodeRange& r, const std::string& msg, const std::string& code)
+  // FIXME, if we make this concurrent, then this should be atomic.
+  static bool error_status{false};
```
Comment (matajoh) [inline src/internal.cc:260]: Is there a way we can make this per-interpreter? Perhaps pass an error object when running the pass (as is done with built-ins presently?)

File: src/interpreter.cc [hunk 2]
```diff
@@ -116,25 +116,39 @@ namespace rego
 
   Node Interpreter::get_errors(const Node& node) const
   {
+    if (!has_err())
+      return {};
+    reset_err();
+
+    Node errorseq = NodeDef::create(ErrorSeq);
```
Comment (matajoh) [inline src/interpreter.cc:123]: If an error object was used, then adding errors to a `Nodes` vector could be something it does when the `err()` method is called.

**Review by matajoh — CHANGES_REQUESTED** [review]

File: src/interpreter.cc [hunk 3]
```diff
@@ -116,6 +116,9 @@ namespace rego
 
   Node Interpreter::get_errors(const Node& node) const
   {
+    if (!node->get_and_reset_contains_error())
```
Comment (matajoh) [inline src/interpreter.cc:119]: You should check if node is null. I believe this may be the source of the segfaults.

Comment (mjp41) [reply src/interpreter.cc]: Fixed. It was a recursive call that was the problem.

**Pull request merged**